server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1993,7 +1993,7 @@ public long getRestoreThrottleTimeInNanos() {
         return restoreRateLimitingTimeInNanos.count();
     }
 
-    protected void assertSnapshotOrGenericThread() {
+    protected void assertSnapshotOrGenericOrStatelessThread() {
```

Comment:
I'd suggest a more intention-revealing name (and one which won't need to change so much in future). How about something like `assertSnapshotAccessPermittedThread()`?